src/datasets/packaged_modules/arrow/arrow.py

```diff
@@ -42,8 +42,12 @@ def _split_generators(self, dl_manager):
             # Infer features if they are stored in the arrow schema
             if self.info.features is None:
                 for file in itertools.chain.from_iterable(files):
-                    with open(file, "rb") as f:
-                        self.info.features = datasets.Features.from_arrow_schema(pa.ipc.open_stream(f).schema)
+                    data_memory_map = pa.memory_map(file)
```

Comment:
Memory mapping is only available for local files, however in streaming mode `file` is a URL (and `open` is extended to work with URLs and returns a valid `f`).

Could you make it work using `f` rather than `data_memory_map ` ?

Reply:
Updated @lhoestq thanks